### ISSUE TITLE
set_monitor with mac80211

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#138]
**Reported by Matteo on 11 Mar 2007 18:43 UTC**

With this patch airodump can put the interface in monitor mode with devices using mac80211
This maybe breaks non-mac80211 devices so it should be applied only after we
figure out how to detect mac80211

**Comment by aircrack-ng** [comment]
**Modified by hirte on 11 Mar 2007 18:43 UTC**

**Comment by aircrack-ng** [comment]
**Comment by Mister X on 11 Mar 2007 18:43 UTC**

That should be put in a function, the code is a bit redundant.
Btw, isn't there something like a file in /proc that could help recognizing Mac802.11 devices?

**Comment by aircrack-ng** [comment]
**Comment by Matteo on 11 Mar 2007 18:43 UTC**

Maybe looking in /sys/class/net/wlan0

**Comment by aircrack-ng** [comment]
**Comment by Matteo on 11 Mar 2007 18:43 UTC**

Got reply from the a mac80211 developer:
----
BTW, I see you are involved in aircrack-ng. For monitoring mac80211 drivers/devices, I recommend using virtual interfaces instead of taking the current interface down and setting it to monitor mode. Pick an interface name, send it to /sys/class/ieee80211/wiphyX/add_iface, set that interface to monitor mode, and bring it up. This allows you to capture frames (and in the future, inject frames) without affecting the current connection, provided there is no channel changing. 
----
```
[echo -n matteo > /sys/class/ieee80211/wiphy0/add_iface
[~](~]#)# ifconfig matteo
matteo    Link encap:Ethernet  HWaddr F1:CA:F1:CA:F1:CA
          BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 b)  TX bytes:0 (0.0 b)
[iwconfig matteo mode monitor
[~](~]#)# iwconfig matteo
matteo    IEEE 802.11g  Mode:Monitor  Channel:0
          RTS thr:off   Fragment thr=2346 B
          Encryption key:off
          Link Quality:0  Signal level:0  Noise level:0
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:0  Invalid misc:0   Missed beacon:0

[~]# airodump-ng matteo
<hehe, it works ;)>
```

**Comment by aircrack-ng** [comment]
**Comment by hirte on 11 Mar 2007 18:43 UTC**

Nice, only took us 5 months to integrate it, but thanks for the information! Fixed in 1.0-dev. ;)
